apollo-router/src/axum_http_server_factory.rs

```diff
@@ -468,34 +469,38 @@ async fn health_check() -> impl IntoResponse {
 }
 
 async fn run_graphql_request(
-    service: Buffer<
-        BoxService<
-            http_compat::Request<graphql::Request>,
-            http_compat::Response<ResponseBody>,
-            BoxError,
-        >,
-        http_compat::Request<graphql::Request>,
-    >,
+    service: BufferedService,
     http_request: Request<graphql::Request>,
 ) -> impl IntoResponse {
     match service.ready_oneshot().await {
         Ok(mut service) => {
             let (head, body) = http_request.into_parts();
 
-            service
+            match service
                 .call(http_compat::Request::from_parts(head, body))
                 .await
-                .map(|response| {
-                    tracing::trace_span!("serialize_response").in_scope(|| response.into_response())
-                })
-                .unwrap_or_else(|e| {
+            {
+                Err(e) => {
                     tracing::error!("router service call failed: {}", e);
                     (
                         StatusCode::INTERNAL_SERVER_ERROR,
                         "router service call failed",
                     )
                         .into_response()
-                })
+                }
+                Ok(mut stream) => match stream.next().await {
+                    None => {
+                        tracing::error!("router service is not available to process request",);
+                        (
+                            StatusCode::SERVICE_UNAVAILABLE,
+                            "router service is not available to process request",
```

Comment:
That s a great idea 🎉

Reply:
that part will change a bit once we actually start sending multiple responses